jujugui/static/gui/src/app/store/env/api.js

```diff
@@ -810,24 +810,50 @@ YUI.add('juju-env-api', function(Y) {
         this.setConnectedAttr('maasServer', null);
         return;
       }
-      this.modelGet(data => {
-        if (data.err) {
-          console.warn('error calling ModelGet API: ' + data.err);
+      this.cloudInfo(data.cloud, data => {
+        if (!data) {
+          console.log('cannot retrieve maas-server from Cloud call');
+          this.setConnectedAttr('maasServer', null);
           return;
         }
-        const maasServer = data.config['maas-server'];
+        if (data.error) {
+          console.warn('error calling Cloud API: ' + data.error);
+          return;
+        }
+        const maasServer = data.cloud.endpoint;
         let value;
         if (maasServer) {
-          value = maasServer.value;
+          value = maasServer;
         } else {
           value = null;
-          console.log('cannot retrieve maas-server from ModelGet call');
+          console.log('cannot retrieve maas-server from Cloud call');
         }
         this.setConnectedAttr('maasServer', value);
       });
     },
 
     /**
+      Fetch the cloud information from the cloud facade. This will include
+      MaaS connection information.
+
+      @method cloudInfo
+      @param {String} cloud The cloud name retrieve information about
+      @param {Function} callback The callback function for handling data
+    */
+    cloudInfo: function(cloud, callback) {
+      // Send the API request.
+      this._send_rpc({
+        type: 'Cloud',
+        request: 'Cloud',
+        params: {
+          entities: [tags.build(tags.CLOUD, cloud)]
+        }
+      }, data => {
+        callback(data.response.results[0]);
```

Comment:
Note that data can also have an error field for global API errors that we are not checking here.